lib/ConvoyFollower/src/App.cpp

```diff
@@ -375,24 +375,34 @@ void App::processPeriodicTasks()
 
     if ((true == m_sendWaypointTimer.isTimeout()) && (true == m_mqttClient.isConnected()))
     {
-        Waypoint      payload;
-        DrivingState& drivingState = DrivingState::getInstance();
-
-        if (false == drivingState.isActive())
-        {
-            /* Not in the correct state. Do nothing. */
-        }
-        else if (false == drivingState.getWaypoint(payload))
+        /* Send debug status. */
+        if (false == m_v2vCommManager.sendStatus(m_latestVehicleData))
         {
-            LOG_WARNING("Failed to get waypoint from driving state.");
+            LOG_WARNING("Status could not be sent.");
         }
-        else if (false == m_v2vCommManager.sendWaypoint(payload))
-        {
-            LOG_WARNING("Waypoint could not be sent.");
-        }
-        else
+
+        if (true == DrivingState::getInstance().isActive())

```

Comment:
Why is this not part of the DrivingState itself?

Reply:
Which part do you mean? Sending the data is not part of the DrivingState because the V2VComm instance belongs to the app class. Checking if the state is active is also done inside the state, but checking it here means it must not be checked after each "get" function call.